### PR TITLE
enh(api) handle resources ACLs in DeleteHost and DeleteService

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Handle ACLs resources on DeleteHost and DeleteService endpoints for non-admin users

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software

## Target serie

- [ ] 21.10.x
- [ ] 22.04.x
- [ ] 22.10.x
- [ ] 23.04.x
- [x] 23.10.x (master)

#### Community contributors & Centreon team

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php [hunk 1]
```diff
@@ -134,6 +135,48 @@ public function exists(int $hostId): bool
...
+    public function existsByAccessGroups(int $hostId, array $accessGroups): bool
+    {
+        if (empty($accessGroups)) {
+            $this->debug('Access groups array empty');
+
+            return false;
+        }
+
+        $accessGroupIds = array_map(
+            fn ($accessGroup) => $accessGroup->getId(),
```
Comment (abuathier) [inline centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php:150]: The higher the phpstan level is, the lower it relies on inference.
Please be the most explicit possible, even if obvious.
```suggestion
            static fn (AccessGroup $accessGroup): int => $accessGroup->getId(),
```

**Review by abuathier — DISMISSED** [review]

**Review by adr-mo — DISMISSED** [review]

**Review by adr-mo — APPROVED** [review]

File: centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php [hunk 2]
```diff
@@ -134,6 +136,48 @@ public function exists(int $hostId): bool
...
+        );
+
+        $concatenator = new SqlConcatenator();
+
+        $concatenator->defineSelect(
+            <<<'SQL'
+                SELECT 1
+                FROM `:db`.host h
+                JOIN `:dbstg`.centreon_acl acl
+                    ON h.host_id = acl.host_id
```
Comment (callapa) [inline centreon/src/Core/Host/Infrastructure/Repository/DbReadHostRepository.php:162]: There's no need to join the host table, you can check directly with the centreon_acl table.

File: centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php [hunk 3]
```diff
@@ -325,7 +325,18 @@ private function getServiceRequestConcatenator(array $accessGroupIds = []): SqlC
...
+                    JOIN `:dbstg`.centreon_acl acl
+                        ON s.service_id = acl.service_id
+                    SQL
+            );
+            $concatenator->appendWhere(
+                <<<'SQL'
+                    WHERE acl.group_id IN (:access_group_ids)
+                    SQL
+            );
+            $concatenator->storeBindValueMultiple(':access_group_ids', $accessGroupIds, \PDO::PARAM_INT);
```
Comment (callapa) [inline centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php:339]: If a non-administrator user has no access groups, no ACLs will be used?

Comment (a-launois) [reply centreon/src/Core/Service/Infrastructure/Repository/DbReadServiceRepository.php:339]: I added the check in the existsByAccessGroups() public method

**Review by adr-mo — APPROVED** [review]

**Pull request merged**